middleware.go

```diff
@@ -0,0 +1,21 @@
+package mux
+
+import "net/http"
+
+type Middleware interface {
```

Comment:
I don't think we should keep this. Single method interfaces can be restrictive; best to let the package user define a wrapper if needed. We end up adding similarly-named methods to the API.

(if there is demand over time, we can re-assess)

Reply:
I still think that adding the `Middleware` interface will help to keep middlewares more standard, and the parallelism with `net/http` `Handler` and `HandlerFunc` helps users to understand the behaviour. This pattern is also seen in other routers and middleware managers (see negroni, for example).

However, if you think it doesn't fit the project, I'll remove it.